nucleus/common/scattered-archive-api/src/test/java/org/glassfish/embeddable/archive/ScatteredArchiveTest.java

```diff
@@ -0,0 +1,123 @@
+        }
+
+        @Override
+        public void addClassPath(File classpath) throws IOException {
+            classpathElements.add(classpath);
+        }
+
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+                "path, '', 1",
+                "path:path2, '', 2",
+                "'jakarta.jakarta-api.jar:/folder/path2:/folder/glassfish-embedded-all.jar', '', 1",
+                "'jakarta.jakarta-api.jar:glassfish-embedded-all.jar', '', 0",
+                "path, 'path', 0",
+                "path111:/folder/path222, 'path.*', 0",
+                "path333:path111:/folder/path222, 'path1.* , path2.*', 1",
+                "'jakarta.jakarta-api.jar:path2:/some/folder/path3:somepath', 'path.', 1",
+                "'', '', 0"
+            })
+    public void addCurrentClasspathUsingExcludes(String classpath, String excludesList, int expectedElementsInClasspath) {
+
+        // GIVEN
+        setCurrentClasspath(classpath);
+        String[] excludesPatterns = excludesList.split(REGEX_COMMA_WITH_SPACE_AROUND);
+        TestableScatteredArchive archive = new TestableScatteredArchive("test", ScatteredArchive.Type.WAR);
+
+        // WHEN
+        archive.addCurrentClassPath(excludesPatterns);
+
+        // THEN
+        assertAll("classpath",
+                () -> assertThat("Number of classpath elements",
+                        archive.classpathElements, hasSize(expectedElementsInClasspath))
+        );
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "path, '', 1",
+        "path:path2, '', 2",
+        "'jakarta.jakarta-api.jar:/folder/path2:/folder/glassfish-embedded-all.jar', '0', 2",
+        "'jakarta.jakarta-api.jar:glassfish-embedded-all.jar', '0,1', 0",
+        "path, '0', 0",
+        "path111:/folder/path222, '0,1', 0",
+        "path111:/folder/path222, '0', 1",
+        "'jakarta.jakarta-api.jar:path2:/some/folder/path3:somepath', '1,2', 2",
+        "'', '', 0"
+    })
+    public void addCurrentClasspathUsingPredicate(String classpath, String excludesIndexes, int expectedElementsInClasspath) {
+
+        // GIVEN
+        var cpElements = setCurrentClasspath(classpath);
+        Predicate<String> exclude = value -> {
+            return Stream.of(excludesIndexes.split(REGEX_COMMA_WITH_SPACE_AROUND))
+                    .filter(s -> !s.isBlank())
+                    .map(Integer::valueOf)
+                    .anyMatch(i -> cpElements[i].equals(value));
+        };
+        TestableScatteredArchive archive = new TestableScatteredArchive("test", ScatteredArchive.Type.WAR);
+
+        // WHEN
+        archive.addCurrentClassPath(exclude);
+
+        // THEN
+        try {
+            assertAll("classpath",
+                    () -> assertThat("Number of classpath elements " + archive.classpathElements,
+                            archive.classpathElements, hasSize(expectedElementsInClasspath))
+            );
+        } catch (AssertionError e) {
+            throw e;
```

Comment:
What is the difference between this and just letting it go without catching?

Reply:
You're right. I probably forgot to remove this after I added the code for debugging purposes. Here's a PR to remove it: https://github.com/eclipse-ee4j/glassfish/pull/24382